src/components/EmbeddedWalletModal/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { Button, Icon, NavButton } from "@stellar/design-system";
+import { useCallback, useEffect } from "react";
+import { createPortal } from "react-dom";
+
+import "./styles.scss";
+
+interface EmbeddedWalletModalProps {
+  visible: boolean;
+  onClose?: () => void;
+  isDismissible?: boolean;
+  title: string;
+  description: string;
+  contentAlign?: "center" | "left";
+  primaryActionLabel?: string;
+  onPrimaryAction?: () => void;
+  isPrimaryActionLoading?: boolean;
+  isPrimaryActionDisabled?: boolean;
+}
+
+const MODAL_OPEN_CLASS_NAME = "modal-open";
+const MODAL_PARENT_ID = "embedded-wallet-modal-root";
+
+export const EmbeddedWalletModal = ({
+  visible,
+  onClose,
+  isDismissible = true,
+  title,
+  description,
+  contentAlign = "center",
+  primaryActionLabel,
+  onPrimaryAction,
+  isPrimaryActionLoading = false,
+  isPrimaryActionDisabled = false,
+}: EmbeddedWalletModalProps) => {
+  const isModalDismissible = Boolean(onClose) && isDismissible;
+
+  const handleClose = useCallback(() => {
+    onClose?.();
+  }, [onClose]);
+
+  const closeOnEscape = useCallback(
+    (event: KeyboardEvent) => {
+      if (!isModalDismissible) {
+        return;
+      }
+      if (event.key === "Escape") {
+        handleClose();
+      }
+    },
+    [handleClose, isModalDismissible],
+  );
+
+  useEffect(() => {
+    if (!visible || !isModalDismissible) {
+      return undefined;
```

Comment:
I think it will return `undefined` if we don't return anything.
```suggestion
      return;
```